src/Authoring/WinRT.SourceGenerator/WinRTRules.cs

```diff
@@ -172,6 +172,11 @@ private static DiagnosticDescriptor MakeRule(string id, string title, string mes
             "Non-array parameter marked with ReadOnlyArray or WriteOnlyArray",
             "Method '{0}' has parameter '{1}' which is not an array, and which has either a "
             + "ReadOnlyArray attribute or a WriteOnlyArray attribute. Windows Runtime does "
-            + "not support marking non-array parameters with ReadOnlyArray or WriteOnlyArray.");
-    } 
+            + "not support marking non-array parameters with ReadOnlyArray or WriteOnlyArray.");
+
+        public static DiagnosticDescriptor UnimplementedInterface = MakeRule(
+            "CsWinRT1027",
+            "Class incorrectly implements an interface",
+            "Class '{0}' does not correctly implement interface '{1}' because member '{2}' is missing or non-public.");
```

Comment:
Remove the period at the end; the messages are either multiple sentences (last ending with period) or one sentence (no period) 

https://github.com/dotnet/roslyn-analyzers/blob/main/src/Microsoft.CodeAnalysis.Analyzers/Microsoft.CodeAnalysis.Analyzers.md#rs1032-define-diagnostic-message-correctly